features/include/pcl/features/impl/scurv.hpp

```diff
@@ -0,0 +1,730 @@
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+#ifndef PCL_FEATURES_IMPL_SCURV_H_
+#define PCL_FEATURES_IMPL_SCURV_H_
+
+#include <pcl/features/scurv.h>
+#include <pcl/common/common.h>
+#include <pcl/features/normal_3d.h>
+#include <pcl/point_types.h>
+#include <pcl/impl/point_types.hpp>
+#include <pcl/features/principal_curvatures.h>
+#include <pcl/ModelCoefficients.h>
+#include <pcl/exceptions.h>
+#include <Eigen/Dense>
+#include <Eigen/Core>
+#include <Eigen/LU>
+#include <Eigen/QR>
+#include <unsupported/Eigen/Splines>
+#include <Eigen/SVD>
+
+#include <iostream>
+#include <string.h>
+#include <math.h>
+#include <numeric>
+#include <algorithm>
+#include <functional>
+#include <typeinfo>
+#include <stdio.h>
+#include <stdlib.h>
+#include <utility>
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT> double
+pcl::SCurVEstimation<PointInT, PointNT>::getNormalizedValue (double x, double low, double high)
+{
+  return (x - low) / (high - low);
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT> double
+pcl::SCurVEstimation<PointInT, PointNT>::getHermiteDerivativeInterpolation(HermitePoint point0, HermitePoint point1, double xi)
+{
+  double h, delta, del0, del1, c2, c3, x;
+  h = point1.x - point0.x;
+  delta = ( point1.f - point0.f ) / h;
+  del0 = ( point0.d - delta ) / h;
+  del1 = ( point1.d - delta ) / h;
+  c2 = -( del0 + del0 + del1 );
+  c3 = ( del0 + del1 ) / h;
+  x = xi - point0.x;
+  return point0.f + x * ( point0.d + x * ( c2 + x * c3 ) );
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////
+template <typename PointInT, typename PointNT> void
+pcl::SCurVEstimation<PointInT, PointNT>::normalizeScale (pcl::PointCloud<pcl::PointNormal> &cloud, int min_range, int max_range)
+{
+  Eigen::Vector4f min_pt, max_pt;
+  Eigen::Vector3f min_coeff, max_coeff;
+  double minimum, maximum;
+
+  pcl::getMinMax3D(cloud, min_pt, max_pt);
+  min_coeff << min_pt(0), min_pt(1), min_pt(2);
+  max_coeff << max_pt(0), max_pt(1), max_pt(2);
+  minimum = min_coeff.minCoeff();
```

Comment:
```suggestion
  const auto minimum = min_pt.head<3>().maxCoeff();
```

and similar for maximum. You can also remove L93 and L94

Reply:
changed